serialization/pickle.ksy

```diff
@@ -0,0 +1,443 @@
+            'opcode::unicode': unicodestringnl
+            'opcode::binunicode': unicodestring4
+            'opcode::binfloat': f8
+            #'opcode::empty_list': no_arg
+            #'opcode::append': no_arg
+            #'opcode::appends': no_arg
+            #'opcode::list': no_arg
+            #'opcode::empty_tuple': no_arg
+            #'opcode::tuple': no_arg
+            #'opcode::tuple1': no_arg
+            #'opcode::tuple2': no_arg
+            #'opcode::tuple3': no_arg
+            #'opcode::empty_dict': no_arg
+            #'opcode::dict': no_arg
+            #'opcode::setitem': no_arg
+            #'opcode::setitem2': no_arg
+            #'opcode::pop': no_arg
+            #'opcode::dup': no_arg
+            #'opcode::mark': no_arg
+            #'opcode::pop_mark': no_arg
+            'opcode::get': decimalnl_short
+            'opcode::binget': u1
+            'opcode::long_binget': u4
+            'opcode::put': decimalnl_short
+            'opcode::binput': u1
+            'opcode::long_binput': u4
+            'opcode::ext1': u1
+            'opcode::ext2': u2
+            'opcode::ext4': u4
+            'opcode::global':  stringnl_noescape_pair
+            #'opcode::reduce': no_arg
+            #'opcode::build': no_arg
+            'opcode::inst': stringnl_noescape_pair
+            #'opcode::obj': no_arg
+            #'opcode::newobj': no_arg
+            'opcode::proto': u1
+            #'opcode::stop': no_arg
+            'opcode::persid': stringnl_noescape
+            #'opcode::binpersid': no_arg
+        doc: |
+          Optional argument for the operation. Data type and length
+          are determined by the value of the opcode.
+
+  decimalnl_short:
+    seq:
+      - id: val
+        type: str
+        encoding: ASCII
+        terminator: 0x0a # "\n"
+    doc: Integer, encoded with the ASCII characters [0-9-].
+  decimalnl_long:
+    seq:
+      - id: val
+        type: str
+        encoding: ASCII
+        terminator: 0x0a # "\n"
+    doc: Integer, encoded with the ASCII chracters [0-9-], followed by 'L'.
+  # TODO Can kaitai express constraint that these are quoted?
+  stringnl:
+    seq:
+      - id: val
+        terminator: 0x0a # "\n"
+    doc: Quoted string, possibly containing Python string escapes.
+  stringnl_noescape:
+    seq:
+      - id: val
+        terminator: 0x0a # "\n"
+    doc: Unquoted string, does not contain string escapes.
+  stringnl_noescape_pair:
+    seq:
+      - id: val1
+        type: stringnl_noescape
+      - id: val2
+        type: stringnl_noescape
+    doc: Pair of unquoted, unescaped strings.
+  unicodestringnl:
+    seq:
+      - id: val
+        type: str
+        encoding: ASCII
+        terminator: 0x0a # "\n"
+    doc: Unquoted string, containing Python Unicode escapes.
+  floatnl:
+    seq:
+      - id: val
+        type: str
+        encoding: ASCII
+        terminator: 0x0a # "\n"
+    doc: Double float, encoded with the ASCII characters [0-9.eE-], or 'nan'.
```

Comment:
How are infinities handled?

Reply:
this type seems to be never referenced

Reply:
> How are infinities handled?

Good question, almost certainly as `inf\n`. I'll confirm, consider other corner cases, and update the doc.

Reply:
> this type seems to be never referenced. pickling a float results in an IEEE float being stored.

`cases:` above is missing a case. Nice spot.